pkg/controller/linstorcsidriver/linstorcsidriver_controller.go

```diff
@@ -143,6 +148,37 @@ func (r *ReconcileLinstorCSIDriver) Reconcile(request reconcile.Request) (reconc
 	return reconcile.Result{}, statusErr
 }
 
+func (r *ReconcileLinstorCSIDriver) reconcileOldSpec(ctx context.Context, csiResource *piraeusv1alpha1.LinstorCSIDriver) error {
+	changed := false
+
+	// ac78c93 2020-05-19 Add options for CSI sidecar container images
```

Comment:
What's the purpose of this comment? We can assume that the reader knows how to use git blame if they are interested in the history. Also, the commit hash will be incorrect after a rebase, which will happen because we use the "rebase and merge" strategy.

Reply:
The main purpose of this comment was to signal _why_ these changes/updates are needed. I envisioned this function to be kind of "append only", so that we have a clean upgrade path for older resources.

You are right that the commit hash would change after rebasing this branch, however as long as it is up-to-date on merge into master, it should never ever change. In my experience `git blame` often needs some manual review because someone refactors whole parts of files and so changes need to be excluded, traced to older files, etc...

However, you are probably right that it is a bit of a brittle setup, I will rework it

Reply:
Replaced with some verbose logging code, which I think can serve the same purpose

Reply:
Just a technical detail:

> as long as it is up-to-date on merge into master, it should never ever change

That holds for how GitLab does fast-forward merges, but GitHub rewrites the committer, so the hashes change.

Reply:
Huh. Never noticed that. But it does make sense